include/spdlog/pattern_formatter-inl.h

```diff
@@ -814,11 +816,31 @@ class short_filename_formatter final : public flag_formatter
         : flag_formatter(padinfo)
     {}
 
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4127) // consider using 'if constexpr' instead
+#endif // _MSC_VER
     static const char *basename(const char *filename)
     {
-        const char *rv = std::strrchr(filename, os::folder_sep);
-        return rv != nullptr ? rv + 1 : filename;
+        // if the size is 2 (1 character + null terminator) we can use the more efficient strrchr
+        // the branch will be elided by optimizations
+        if (sizeof(os::folder_seps) == 2)
```

Comment:
Please post a dedicated pr for this optimization.

Reply:
This change doesn't make sense without the rest of the folder separator changes